src/object/obj_enum.c

```diff
@@ -708,15 +708,10 @@ static void
 clear_iod(daos_iod_t *iod, d_sg_list_t *sgl, int *recxs_cap)
 {
 	daos_iov_free(&iod->iod_name);
-	if (iod->iod_recxs != NULL)
-		D_FREE(iod->iod_recxs);
+	D_FREE(iod->iod_recxs);
```

Comment:
This function prototype is odd, and it's only called in two places (although it is static), would it not be possible to just do the assignment of *rexcx_cap in the caller or make this a macro?

Reply:
@wangdi1, what do think about this?

Reply:
yes, *rexcx_cap assignment seems irrelevant,  so move it outside and make this as a macro makes sense to me.